pkg/registry/service/rest.go

```diff
@@ -402,3 +477,16 @@ func externalLoadBalancerNeedsUpdate(oldService, newService *api.Service) bool {
 	}
 	return false
 }
+
+// Determine if the service needs a cluster-wide port assigned
+func (rs *REST) needsServicePort(service *api.Service) bool {
+	if service.Spec.CreateExternalLoadBalancer {
+		balancer, ok := rs.cloud.TCPLoadBalancer()
```

Comment:
why create a balancer just to throw it away?  Why not cloud.LoadBalancerNeedsHostPort() or something?